anesthetic/utils.py

```diff
@@ -89,6 +90,148 @@ def quantile(a, q, w=None, interpolation='linear'):
     return quant
 
 
+def sample_cdf(samples, inverse=False, interpolation='linear'):
+    """Sample the empirical cdf for a 1d array."""
+    samples = np.array(samples)
+    samples.sort()
+    ngaps = len(samples)-1
+    gaps = np.random.dirichlet(np.ones(ngaps))
+    cdf = np.array([0, *np.cumsum(gaps)])
+    assert np.isclose(cdf[-1], 1, atol=1e-9, rtol=1e-9), \
+        "Error: CDF does not reach 1 but "+str(cdf[-1])
+    # Set the last element (tested to be approx 1)
+    # to exactly 1 to avoid interpolation errors
+    cdf[-1] = 1
```

Comment:
Would it be more appropriate to normalise as follows?
```python
cdf /= cdf[-1]
```

Or would it be better to handle bound errors within `interp1d`? (see below)

Reply:
Honestly it doesn't matter since ` atol=1e-9`, this is just dealing with python float precision. So it's stick to the simpler version to avoid looking like we're doing some actual normalisation